tools/js2c.py

```diff
@@ -256,10 +264,19 @@ def JS2C(source, target):
     lines = ExpandConstants(lines, consts)
     lines = ExpandMacros(lines, macros)
 
+    deprecatedDeps = None
+
     # On Windows, "./foo.bar" in the .gyp file is passed as "foo.bar"
     # so don't assume there is always a slash in the file path.
     if '/' in name or '\\' in name:
-      name = '/'.join(re.split('/|\\\\', name)[1:])
+      split = re.split('/|\\\\', name)
+      if split[0] == 'deps':
+        if split[1] == 'node-inspect' or split[1] == 'v8':
+          deprecatedDeps = split[1:]
+        split = ['internal'] + split
```

Comment:
This maps `deps/foo/bar.js` to `internal/deps/foo/bar.js`.

I don't think you have to special-case for `node-inspect` or `v8` because anything from deps/ should pretty much be internal by default.  ISTM it can be simplified to his:
```python
if split[0] === 'deps':
  split[0] = 'internal'
else:
  split = split[1:]
```
Could be simplified further by just matching on `name`:
```python
name = '/'.join(re.split('/|\\\\', name))
if name.startswith('lib/'):
  name = name[len('lib/'):]
elif name.startswith('deps/'):
  deprecated_deps, name = name, 'internal/' + name[len('deps/'):]
```
Bonus points: you won't have to `'/'.join(...)` below.

Reply:
I think I misread the intent of this code - it's prefixing with `internal/` unconditionally but stashing the original away.  Could still be simplified, though:
```python
name = '/'.join(re.split('/|\\\\', name))
if name.startswith('deps/node-inspect/') or name.startswith('deps/v8/'):
  deprecated_deps = name[len('deps/'):]
  name = 'internal/' + name
elif name.startswith('lib/'):
  name = name[len('lib/'):]
```

Reply:
@bnoordhuis Not only that, but `internal/` should also be added to any other file in `deps/` that do not match `node-inspect` and `v8` to prevent them from being visible to userland, thus the code in its current form.